iocore/hostdb/HostDB.cc

```diff
@@ -634,19 +634,22 @@ HostDBProcessor::getby(Continuation *cont, const char *hostname, int len, sockad
   // Attempt to find the result in-line, for level 1 hits
   //
   if (!aforce_dns) {
+    MUTEX_TRY_LOCK(lock, cont->mutex, thread);
+    if (!lock.is_locked()) {
+      goto Ldelay;
```

Comment:
I'm very much not fond of `goto`. In this case, I'd be tempted to change the `do;while(loop)` into a `while (loop)`. E.g. something like
```
MUTEX_TRY_LOCK(...);
bool loop = lock.is_locked();
while (loop) {
  loop = false; // ...
  // ...
}
```

Reply:
Updated per your suggestion.